File: C++/API/Treehopper.Libraries/src/Sensors/Inertial/Accelerometer.cpp
#include "Sensors/Inertial/Accelerometer.h"

namespace Treehopper {
    namespace Libraries {
        namespace Sensors {
            namespace Inertial {
                vector3_t &Accelerometer::accelerometer() {
                    if (autoUpdateWhenPropertyRead) update();
                    return _accelerometer;
                }
            }
        }
    }
}

File: Java/api/treehopper.libraries/src/main/java/io/treehopper/libraries/sensors/inertial/Accelerometer.java
package io.treehopper.libraries.sensors.inertial;

import com.badlogic.gdx.math.Vector3;

public abstract class Accelerometer implements IAccelerometer {
    protected boolean autoUpdateWhenPropertyRead;
    protected Vector3 accelerometer;

    @Override
    public Vector3 getAccelerometer() {
        if (autoUpdateWhenPropertyRead)
            update();

        return accelerometer;
    }

    @Override
    public boolean isAutoUpdateWhenPropertyRead() {
        return autoUpdateWhenPropertyRead;
    }

    @Override
    public void setAutoUpdateWhenPropertyRead(boolean value) {
        autoUpdateWhenPropertyRead = value;
    }
}
